matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/EventRelationsAggregationProcessor.kt

```diff
@@ -532,6 +546,49 @@ internal class EventRelationsAggregationProcessor @Inject constructor(
         }
     }
 
+    private fun handleLiveLocation(realm: Realm,
```

Comment:
In my opininion, we clearly need to stop adding code to this class. It is too big. To separate easily the code, I propose using delegation pattern. The idea is:

- create a dedicated package named `aggregation.livelocation`. 
- In this package, create an interface as follow:
```kotlin
interface LiveLocationAggregationProcessor {
    fun handleLiveLocation(realm: Realm,
                           event: Event,
                           content: MessageLiveLocationContent,
                           roomId: String,
                           isLocalEcho: Boolean)
}
```
- In the same package create the class to implement the aggregation (moved from `EventRelationsAggregationProcessor`)
- Finally make `EventRelationsAggregationProcessor` implements the `LiveLocationAggregationProcessor` interface using delegation:
```kotlin
EventRelationsAggregationProcessor(...) : LiveLocationAggregationProcessor by DefaultLiveLocationAggregationProcessor() {...}
```

This way we have a class dedicated to aggregating a single event type and it is easier to add unit tests on it.
What do you think?

Reply:
Right, done. It is also easy to test now.